presto-hive/src/main/java/com/facebook/presto/hive/HiveMetadata.java

```diff
@@ -601,6 +604,60 @@ public ConnectorTableMetadata getTableMetadata(ConnectorSession session, Connect
         return getTableMetadata(session, ((HiveTableHandle) tableHandle).getSchemaTableName());
     }
 
+    @VisibleForTesting
+    static Function<HiveColumnHandle, ColumnMetadata> columnMetadataGetter(Table table, TypeManager typeManager, ColumnConverter columnConverter)
+    {
+        ImmutableList.Builder<String> columnNames = ImmutableList.builder();
+        table.getPartitionColumns().stream().map(Column::getName).forEach(columnNames::add);
+        table.getDataColumns().stream().map(Column::getName).forEach(columnNames::add);
+        List<String> allColumnNames = columnNames.build();
+        if (allColumnNames.size() > Sets.newHashSet(allColumnNames).size()) {
+            throw new PrestoException(HIVE_INVALID_METADATA,
+                    format("Hive metadata for table %s is invalid: Table descriptor contains duplicate columns", table.getTableName()));
+        }
+
+        List<Column> tableColumns = table.getDataColumns();
+        ImmutableMap.Builder<String, Optional<String>> builder = ImmutableMap.builder();
+        for (Column field : concat(tableColumns, table.getPartitionColumns())) {
+            if (field.getComment().isPresent() && !field.getComment().get().equals("from deserializer")) {
+                builder.put(field.getName(), field.getComment());
+            }
+            else {
+                builder.put(field.getName(), Optional.empty());
+            }
+        }
+        // add hidden columns
+        builder.put(PATH_COLUMN_NAME, Optional.empty());
+        if (table.getStorage().getBucketProperty().isPresent()) {
+            builder.put(BUCKET_COLUMN_NAME, Optional.empty());
+        }
+
+        builder.put(FILE_SIZE_COLUMN_NAME, Optional.empty());
+        builder.put(FILE_MODIFIED_TIME_COLUMN_NAME, Optional.empty());
+
+        Map<String, Optional<String>> columnComment = builder.build();
+
+        ImmutableMap.Builder<String, Optional<String>> typeMetadataBuilder = ImmutableMap.builder();
+
+        for (Column field : concat(tableColumns, table.getPartitionColumns())) {
+            if (field.getTypeMetadata().isPresent() && !field.getTypeMetadata().get().isEmpty()) {
```

Comment:
We don't have to check `field.getTypeMetadata().get().isEmpty()`. If we have an empty, then it must be a corrupted type, which would be a much worse situation.